### PR TITLE
rpcserver: Add handleGet{NetworkHashPS,MiningInfo} tests.

### PR DESCRIPTION
**Pull request by JoeGruffins**
This adds tests for `handleGetNetworkHashPS` and `handleGetMiningInfo`.
    
Because the test blockchain `HeaderByHash` must return different timestamps, use a function for `testRPCChain.headerByHash` that can return varying results.

### CODE REVIEW

**Comment by JoeGruffins** [comment]
```
$ go test -run="TestHandleGetNetworkHashPS" -coverprofile=cov.out && go tool cover -func=cov.out | grep -E "handleGetNetworkHashPS"
PASS
coverage: 1.7% of statements
ok      github.com/decred/dcrd/internal/rpcserver       0.015s
github.com/decred/dcrd/internal/rpcserver/rpcserver.go:2559:            handleGetNetworkHashPS          100.0%

$ go test -run="TestHandleGetMiningInfo" -coverprofile=cov.out && go tool cover -func=cov.out | grep -E "handleGetMiningInfo"
PASS
coverage: 1.9% of statements
ok      github.com/decred/dcrd/internal/rpcserver       0.014s
github.com/decred/dcrd/internal/rpcserver/rpcserver.go:2515:            handleGetMiningInfo             90.0%
```

File: internal/rpcserver/rpcserverhandlers_test.go
Comment (davecgh) [inline internal/rpcserver/rpcserverhandlers_test.go:6243]: The header in the message really isn't needed.  Test failures always log the line they came from and the test function that was running, so it doesn't provide any extra information.  That said, it's not a big deal, so I'll approve anyway.  Feel free to change it if you want.

**Review by davecgh — APPROVED** [review]

File: internal/rpcserver/rpcserverhandlers_test.go
Comment (dnldd) [inline internal/rpcserver/rpcserverhandlers_test.go:4511]: I think the test names associated with errors could provide more detail if the they captured the context of the error generated. In this case a name like `"handleGetMiningInfo: invalid network hashes per sec"` would be a better fit, thoughts?

File: internal/rpcserver/rpcserverhandlers_test.go
Comment (dnldd) [inline internal/rpcserver/rpcserverhandlers_test.go:4597]: same here.

File: internal/rpcserver/rpcserverhandlers_test.go
Comment (dnldd) [inline internal/rpcserver/rpcserverhandlers_test.go:4608]: same here.

**Comment by JoeGruffins** [comment]
Will implement suggestions.

File: internal/rpcserver/rpcserverhandlers_test.go
Comment (JoeGruffins) [reply internal/rpcserver/rpcserverhandlers_test.go:4511]: Ok!

**Comment by davecgh** [comment]
Holding off on merge for the aforementioned updates.

**Review by davecgh — APPROVED** [review]
Updates look good.  Will wait for @dnldd to approve before merging to make sure he feels everything is addressed.

**Comment by JoeGruffins** [comment]
ah, put the spelling changes in the wrong commit, one second

**Review by dnldd — APPROVED** [review]

**Pull request merged**
